anker/src/state.rs

```diff
@@ -339,20 +347,29 @@ impl Anker {
         );
             return Err(AnkerError::WrongSplTokenSwapParameters.into());
         }
+
+        // Get mint from tokens, depending on the order.
+        let (pool_st_sol_mint, pool_ust_mint) =
+            if &token_swap.token_a_mint == accounts.st_sol_mint.key {
```

Comment:
We should merge this with the earlier check, and resolve which one is A and which one is B only once. Right now what you can do is pass in the UST mint for the `st_sol_mint` argument, and it would not cause a failure.

Reply:
Thanks! Modified it.